File: freak-fights-fan/shared/features/images/queries/GetAllImages.ts
import { SortOrder } from "../../../abstractions/SortOrder";
import { IPagedQuery } from "../../../abstractions/IPagedQuery";
import { ISortedQuery } from "../../../abstractions/ISortedQuery";

export namespace GetAllImages {
    export interface Query extends IPagedQuery, ISortedQuery {
        page: number;
        pageSize: number;
        sortColumn: string;
        sortOrder: SortOrder;
    }
}

File: FreakFightsFan.Shared/Features/Images/Queries/GetAllImages.cs
using FluentValidation;
using FreakFightsFan.Shared.Abstractions;
using FreakFightsFan.Shared.Features.Images.Responses;
using MediatR;

namespace FreakFightsFan.Shared.Features.Images.Queries;

public static class GetAllImages
{
    public class Query : IRequest<PagedList<ImageDto>>, IPagedQuery, ISortedQuery
    {
        public int Page { get; set; }
        public int PageSize { get; set; }
        public string SortColumn { get; set; }
        public SortOrder SortOrder { get; set; }
    }

    public class Validator : AbstractValidator<Query>
    {

    }
}